File: src/app/connect/model/connect/runtime/RuntimeOptionsResponse.ts
import {RdfsClass} from '../../../../platform-services/tsonld/RdfsClass';
import {RdfId} from '../../../../platform-services/tsonld/RdfId';
import {RdfProperty} from '../../../../platform-services/tsonld/RdfsProperty';
import {StaticProperty} from "../../StaticProperty";
import {RuntimeOptionsRequest} from "./RuntimeOptionsRequest";
import {Option} from "../../Option";

@RdfsClass('sp:RuntimeOptionsResponse')
export class RuntimeOptionsResponse extends RuntimeOptionsRequest {


    @RdfProperty('sp:hasRequestId')
    public requestId: string;

    @RdfProperty('sp:hasOption')
    public options: Option[] = [];


    constructor() {
        super();
        //this.id = "http://streampipes.org/transformation_rule/" + Math.floor(Math.random() * 10000000) + 1;
        //this.runtimeKey = runtimeKey;
    }

}

File: streampipes-model/src/main/java/org/streampipes/model/runtime/RuntimeOptionsResponse.java
/*
 * Copyright 2018 FZI Forschungszentrum Informatik
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */
package org.streampipes.model.runtime;

import org.streampipes.empire.annotations.RdfProperty;
import org.streampipes.empire.annotations.RdfsClass;
import org.streampipes.model.staticproperty.Option;
import org.streampipes.vocabulary.StreamPipes;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToMany;

@RdfsClass(StreamPipes.RUNTIME_OPTIONS_RESPONSE)
@Entity
public class RuntimeOptionsResponse extends RuntimeOptionsRequest {

  @OneToMany(fetch = FetchType.EAGER,
          cascade = {CascadeType.ALL})
  @RdfProperty(StreamPipes.HAS_OPTION)
  private List<Option> options;

  public RuntimeOptionsResponse() {
    super();
  }

  public RuntimeOptionsResponse(RuntimeOptionsRequest request, List<Option> options) {
    super();
    this.requestId = request.getRequestId();
    this.options = options;
  }

  public List<Option> getOptions() {
    return options;
  }

  public void setOptions(List<Option> options) {
    this.options = options;
  }
}
